src/Stratis.Bitcoin.IntegrationTests/Transactions/TransactionWithNullDataSteps.cs

```diff
@@ -112,6 +112,7 @@ private void the_block_is_mined()
         {
             this.blockWithOpReturnId = this.senderNode.GenerateStratisWithMiner(1).Single();
             this.senderNode.GenerateStratisWithMiner(1);
+            TestHelper.WaitLoop(() => TestHelper.IsNodeSynced(this.senderNode));
```

Comment:
do you need this one?
You're waiting for `senderNode` i nthe next line.

Reply:
Yes, I have found out that AreNodesSynced only check if 2 nodes are in sync with each other, but not if they are up to date, this test was working nicely on my slow PC, then I noticed it was failing on CI build and a faster PC. We have prepared something with Andy to avoid people getting caught by that in another review

Reply:
https://github.com/stratisproject/StratisBitcoinFullNode/pull/1304